app/shared/dom-hooks/use-scroll-state.ts

```diff
@@ -0,0 +1,62 @@
+import { useEffect } from "react";
+import mitt from "mitt";
+import noop from "lodash.noop";
```

Comment:
would be a lot of noops all over the place, ultimately doesn't matter, but may be important for referential equality in some cases, I think generally any component that is reusable should have noop as a default for callback props, so noop could be a highly used function and in this case I feel like its worth importing? wdyt?

Reply:
Yeah it’s fine for sure but I prefer no-lib solution for simple cases like this. Reference doesn’t make a difference here since it’s not in the render.